pkg/chain/ethereum/ethereum.go

```diff
@@ -43,6 +50,12 @@ type Chain struct {
 	// nonce. Serializing submission ensures that each nonce is requested after
 	// a previous transaction has been submitted.
 	transactionMutex *sync.Mutex
+
+	tokenStaking *contract.TokenStaking
+}
+
+func (ec *Chain) RolesOf(stakingProvider chain.Address) (owner, beneficiary, authorizer chain.Address, err error) {
+	return chain.Address(""), chain.Address(""), chain.Address(""), nil
 }
```

Comment:
I would consider putting this function in a separate file, `threshold.go`.

Reply:
I would also consider returning `bool` to see if the roles are defined. This is similar to `OperatorToStakingProvider` where we return `bool` to see if the operator is non-zero. The chain implementation knows what it means the address is zero, the consuming code does not.

In case of this function, I think it's enough to check if the staking provider is non-zero.

Reply:
Done in 4954e861ac7b562c2593429e56fc922e5f861012.